src/python/zquantum/core/measurement.py

```diff
@@ -435,28 +446,69 @@ def check_parity(
     return result
 
 
+def check_parity_of_vector(
+    bitstrings_vector: np.ndarray, marked_qubits: Iterable[int]
+) -> np.ndarray:
+    """Determine if the marked qubits have even parity for each bitstring in the given
+        vector.
```

Comment:
I would add a comment here that it's similar in functionality to `check_parity` but much faster as it makes use of vectorization.

Reply:
Fixed in [df4fd65](https://github.com/zapatacomputing/z-quantum-core/pull/452/commits/df4fd654ebd2a866f21c8f0352caabf79c1ec68c)